backend/de.metas.cucumber/src/test/java/de/metas/cucumber/stepdefs/externalreference/S_ExternalReference_StepDef.java

```diff
@@ -99,6 +111,65 @@ public void metasfresh_contains_s_external_reference(@NonNull final DataTable da
 		}
 	}
 
+	@And("remove external reference if exists:")
+	public void remove_external_reference_if_exists(@NonNull final DataTable dataTable)
+	{
+		final List<Map<String, String>> tableRows = dataTable.asMaps(String.class, String.class);
+		for (final Map<String, String> tableRow : tableRows)
+		{
+			removeExternalReferenceIfExists(tableRow);
+		}
+	}
+
+	@Given("metasfresh contains S_ExternalReferences")
+	public void theUserAddsBpartnerExternalReference(@NonNull final DataTable dataTable)
+	{
+		final List<Map<String, String>> dataTableEntries = dataTable.asMaps();
+
+		final IQueryOrderBy orderBy =
+				queryBL.createQueryOrderByBuilder(I_C_BPartner.class)
+						.addColumn(I_C_BPartner.COLUMN_C_BPartner_ID)
+						.createQueryOrderBy();
+
+		final List<JsonMetasfreshId> bPartnerIds = queryBL.createQueryBuilder(I_C_BPartner.class)
+				.addOnlyActiveRecordsFilter()
+				.create()
+				.setOrderBy(orderBy)
+				.list()
+				.stream()
+				.map(bPartner -> JsonMetasfreshId.of(bPartner.getC_BPartner_ID())).collect(Collectors.toList());
+
+		dataTableEntries.forEach(dataTableEntry -> {
```

Comment:
No offense, but I wonder what you folks see in `.forEach`...i mean, how is is better than a normal for-loop at this point?

Reply:
Changed to normal for-each loop